runtime/src/in_mem_accounts_index.rs

```diff
@@ -49,6 +49,9 @@ pub struct InMemAccountsIndex<T: IndexValue> {
     stop_evictions: AtomicU64,
     // set to true while this bin is being actively flushed
     flushing_active: AtomicBool,
+
+    /// info to streamline initial index generation
+    startup_info: RwLock<StartupInfo<T>>,
```

Comment:
For large account counts, we cannot accumulate inserts in memory - they have to be written as fast as possible to disk to avoid oom.

Reply:
I could use a mutex here. It is only written to/swapped during startup, but we do assert it is empty, which is a read, but this should only every happen once per bin per startup and this will never happen while anyone wants to write to it ever again.